src/components/taskDetails/TaskDates.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import Details from './Details';
+import styles from './task-details.module.scss';
+import { TASK_EXTENSION_REQUEST_URL } from '@/constants/url';
+import convertTimeStamp from '@/helperFunctions/convertTimeStamp';
+
+interface TaskDatesProps {
+    isEditing: boolean;
+    isUserAuthorized: boolean;
+    startedOn: string;
+    endsOn: number;
+    newEndOnDate: string;
+    setNewEndOnDate: (date: string) => void;
+    handleBlurOfEndsOn: () => void;
+    isExtensionRequestPending: boolean;
+    taskId: string;
+}
+
+const TaskDates: React.FC<TaskDatesProps> = ({
+    isEditing,
+    isUserAuthorized,
+    startedOn,
+    endsOn,
+    newEndOnDate,
+    setNewEndOnDate,
+    handleBlurOfEndsOn,
+    isExtensionRequestPending,
+    taskId,
+}) => {
+    const getExtensionRequestLink = (
+        taskId: string,
+        isExtensionRequestPending?: boolean
+    ) => {
+        return isExtensionRequestPending
+            ? `${TASK_EXTENSION_REQUEST_URL}?&q=${encodeURIComponent(
+                  `taskId:${taskId},status:PENDING`
+              )}`
+            : null;
+    };
+
+    const formattedEndsOn = endsOn ? convertTimeStamp(endsOn) : 'TBD';
+
+    return (
+        <>
+            <div className={styles.inputContainer}>
+                <Details detailType={'Started On'} value={startedOn} />
+            </div>
+            <div className={styles.inputContainer}>
+                <Details
```

Comment:
- `Details` name is very generic can we change the name to indicate what the component does?

Reply:
The `Details` component is already integrated into various parts of the codebase. Changing its name would require extensive updates across multiple files, which could introduce unnecessary complexity and increase the risk of errors. The name `Details` describes the component's purpose, which is to display various types of task details. Given its widespread usage and the clear intent of its current name, I believe it's more practical to retain the existing name.